addons/udes_stock/models/stock_move_line.py

```diff
@@ -336,21 +336,24 @@ def _filter_by_products_info(self, products_info):
     def get_package_move_lines(self, package):
         """ Get move lines of package when package is a package or
             a parent package, and to handle swapping packages in
-            case the expected_package_name entry is included in
+            case the expected_package_names entry is included in
             the context.
 
         """
         Package = self.env['stock.quant.package']
 
         package.ensure_one()
         move_lines = None
-        expected_package_name = self.env.context.get('expected_package_name')
+        expected_package_names = self.env.context.get('expected_package_names')
         picking = self.mapped('picking_id')
 
-        if expected_package_name is not None \
-           and expected_package_name != package.name:
-            expected_package = Package.get_package(expected_package_name)
-            move_lines = picking.maybe_swap(package, expected_package)
+        if expected_package_names is not None \
+           and package.name not in expected_package_names:
+
+            expected_packages = Package.search([
+                ("name", "in", expected_package_names),
```

Comment:
is this right? just name? =s

Reply:
oh expected_package_names is a recordset

Reply:
It is a list of names, otherwise it wouldn't work as a search parameter.